app/src/main/java/io/github/wulkanowy/data/repositories/grade/GradeRepository.kt

```diff
@@ -43,7 +43,24 @@ class GradeRepository @Inject constructor(
                         local.getGradesSummary(semester).toSingle(emptyList())
                             .doOnSuccess { old ->
                                 local.deleteGradesSummary(old.uniqueSubtract(newSummary))
-                                local.saveGradesSummary(newSummary.uniqueSubtract(old))
+                                local.saveGradesSummary(newSummary.uniqueSubtract(old)
+                                    .onEach { summary ->
+                                        val oldSummary = old.find { oldSummary ->
+                                            oldSummary.subject == summary.subject &&
+                                                oldSummary.studentId == summary.studentId &&
+                                                oldSummary.semesterId == summary.semesterId
+                                        }
+                                        if (summary.predictedGrade.isEmpty()) {
+                                            summary.isPredictedGradeNotified = true
+                                        } else if (notify && oldSummary != null && oldSummary.predictedGrade != summary.predictedGrade) {
+                                            summary.isPredictedGradeNotified = false
+                                        }
```

Comment:
Możesz też spróbować zrobić coś takiego:

```suggestion
                                        summary.isPredictedGradeNotified = when {
                                            summary.predictedGrade.isEmpty() -> true
                                            notify && oldSummary?.predictedGrade != summary.predictedGrade -> false
                                            else -> true
                                    }
```